zboxcore/zboxutil/http.go

```diff
@@ -741,7 +741,21 @@ func MakeSCRestAPICall(scAddress string, relativePath string, params map[string]
 func HttpDo(ctx context.Context, cncl context.CancelFunc, req *http.Request, f func(*http.Response, error) error) error {
 	// Run the HTTP request in a goroutine and pass the response to f.
 	c := make(chan error, 1)
-	go func() { c <- f(Client.Do(req.WithContext(ctx))) }()
+	go func() {
+		var err error
+		// indefinitely try if io.EOF error occurs. As per some research over google
+		// it occurs when client http tries to send byte stream in connection that is
+		// closed by the server
+		for {
```

Comment:
use context to break this loop and terminate goroutine properly

```
for {
    select {
        case <-ctx.Done():
             return
        default:
             //your code
    }
}
```

Reply:
Context is already in use inside `client.Do` function. Only for EOF error it will continue. If it frequently shows up EOF error then it will anyway break when context timeout occurs

Reply:
ok, it should be enough